src/server/qgswfsserver.cpp

```diff
@@ -432,9 +432,14 @@ int QgsWFSServer::getFeature( QgsRequestHandler& request, const QString& format
   {
     QDomElement docElem = doc.documentElement();
     if ( docElem.hasAttribute( "maxFeatures" ) )
+    {
       maxFeatures = docElem.attribute( "maxFeatures" ).toLong();
+      maxFeat = maxFeatures;
```

Comment:
The mix between maxFeatures and maxFeat is a bit messy in this method. I'd suggest replacing maxFeatures with a bool `hasFeatureLimit`.